scripts/test/Muon/load_run_widget/loadrun_presenter_single_file_test.py

```diff
@@ -0,0 +1,291 @@
+            thread_model._thread.wait()
+            QT_APP.processEvents()
+
+    def setUp(self):
+        # Store an empty widget to parent all the views, and ensure they are deleted correctly
+        self.obj = QtGui.QWidget()
+
+        self.data = MuonLoadData()
+        self.view = LoadRunWidgetView(parent=self.obj)
+        self.model = LoadRunWidgetModel(self.data)
+        self.presenter = LoadRunWidgetPresenter(self.view, self.model)
+
+        self.view.warning_popup = mock.Mock()
+        self.presenter.enable_multiple_files(False)
+        self.presenter.set_current_instrument("EMU")
+
+        patcher = mock.patch('Muon.GUI.Common.load_run_widget.model.load_utils')
+        self.addCleanup(patcher.stop)
+        self.load_utils_patcher = patcher.start()
+
+        self.load_single_run()
+
+    def tearDown(self):
+        self.obj = None
+
+    def load_single_run(self):
+        self._loaded_run = 1234
+        self._loaded_filename = "EMU00001234.nxs"
+        self._loaded_workspace = [1, 2, 3]
+
+        self.load_utils_patcher.load_workspace_from_filename = mock.Mock(
+            return_value=(self._loaded_workspace, self._loaded_run, self._loaded_filename))
+        self.view.set_run_edit_text(str(self._loaded_run))
+        self.presenter.handle_run_changed_by_user()
+        self.wait_for_thread(self.presenter._load_thread)
+
+    def mock_model_to_load_single_run(self):
+        self._loaded_run = 1234
+        self._loaded_filename = "EMU00001234.nxs"
+        self._loaded_workspace = [1, 2, 3]
+
+        self.model.load_workspace_from_filename = mock.Mock(
+            side_effect=zip((self._loaded_workspace, self._loaded_filename, self._loaded_run)))
+        self.view.set_run_edit_text(str(self._loaded_run))
+
+    def assert_model_has_not_changed(self):
+        self.assertEqual(self.model.loaded_workspaces, [self._loaded_workspace])
+        self.assertEqual(self.model.loaded_runs, [self._loaded_run])
+        self.assertEqual(self.model.loaded_filenames, [self._loaded_filename])
+
+    def assert_view_has_not_changed(self):
+        self.assertEqual(self.view.get_run_edit_text(), str(self._loaded_run))
+
+    def load_failure(self):
+        raise ValueError("Error text")
+
+    def mock_model_to_throw(self):
+        self.model.load_workspace_from_filename = mock.Mock(side_effect=self.load_failure)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS : Test the increment/decrement buttons in single file mode (can only load one run at a time)
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @run_test_with_and_without_threading
+    def test_that_decrement_run_attempts_to_load_the_correct_run(self):
+        new_filename = "EMU00001233.nxs"
+        load_call_count = self.load_utils_patcher.load_workspace_from_filename.call_count
```

Comment:
should this copy the values explicitly? As it seems to be part of comparison below

Reply:
new_filename is I think the expected value of the filename after it has been decremented. At the start of the test the filename is `EMU00001234.nxs`